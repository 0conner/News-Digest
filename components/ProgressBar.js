import React, {useRef, useState, useEffect} from 'react';
import { View, StyleSheet, Animated } from 'react-native';


function useInterval(callback, delay) {
  const savedCallback = useRef();
  useEffect(() => {
    savedCallback.current = callback;
  }, [callback]);

  useEffect(() => {
    function tick() {
      savedCallback.current();
    }
    if (delay !== null) {
      let id = setInterval(tick, delay);
      return () => clearInterval(id);
    }
  }, [delay]);
}


const ProgressBar = () => {
  let animation = useRef(new Animated.Value(0));
  const [progress, setProgress] = useState(0);
  useInterval(() => {
    if(progress < 100) {
      setProgress(progress + 5);
    }
  }, 0);

  useEffect(() => {
    Animated.timing(animation.current, {
      toValue: progress,
      duration: 4200
    }).start();
  },[progress])

  const width = animation.current.interpolate({
    inputRange: [0, 100],
    outputRange: ["0%", "100%"],
    extrapolate: "clamp"
  })
  return (
    <View style={styles.container}>
      <View style={styles.progressBar}>
        <Animated.View style={[StyleSheet.absoluteFill], {backgroundColor: "#8BED4F", width }}/>
      </View>
    </View>
  );
}

export default ProgressBar;

const styles = StyleSheet.create({
  container: {
    flexDirection: 'column',
    marginTop:5,
    alignItems: 'center',
    backgroundColor: '#ecf0f1',
    padding: 8,
    marginTop:30
  },
  progressBar: {
    flexDirection: 'row',
    height: 10,
    width: '100%',
    backgroundColor: 'white',
    borderColor: '#000',
    borderWidth: 2,
    borderRadius: 5
  }
});
