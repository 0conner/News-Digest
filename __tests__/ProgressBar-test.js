import 'react-native';
import React from 'react';
import ProgressBar from '../components/ProgressBar';
import renderer from 'react-test-renderer';

test('Details snapShot', ()=>{
    const snap=renderer.create(
        <ProgressBar />
    ).toJSON();
expect(snap).toMatchSnapshot();
});