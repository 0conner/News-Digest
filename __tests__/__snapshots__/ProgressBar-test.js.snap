// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details snapShot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ecf0f1",
      "flexDirection": "column",
      "marginTop": 30,
      "padding": 8,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "borderColor": "#000",
        "borderRadius": 5,
        "borderWidth": 2,
        "flexDirection": "row",
        "height": 10,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#8BED4F",
          "width": "0%",
        }
      }
    />
  </View>
</View>
`;
